learningequality/kolibri
# Upgrade KDS to v5.0.0-rc1, introduce KDS live region and use it at one place

## Summary

- Upgrades KDS from `4.3.2` to `5.0.0-rc1`
  - No breaking changes 
  - Related release notes
    - https://github.com/learningequality/kolibri-design-system/releases/tag/v4.4.0
    - https://github.com/learningequality/kolibri-design-system/releases/tag/v5.0.0-rc1
- Utilizes `useKLiveRegion` at one place
- Also fixes a bug when the storage notification banner wasn't visible in cases when the problem it was supposed to communicate was present from the very beginning of loading Kolibri (not sure if that could actually happen, but the fix at least removes the dependency on how other parts of logic work)

## References

- Kolibri counterpart of https://github.com/learningequality/kolibri-design-system/pull/687
- Related https://github.com/learningequality/kolibri/issues/12326
  - I will close this issue but not in the scope of this PR. I'd rather work on remaining regions gradually. Quite frequently, they currently don't work like expected and some clarifications will be needed - I don't have bandwith to do it in the scope of this work.

## Reviewer guidance

**Aug 7 update: below won't work as I cleaned up temporary changes that allowed for testing.**

**Code review**
- Skip reviewing https://github.com/learningequality/kolibri/pull/12475/commits/6a573c8ea803c6a4d619933bb1fc8bb9d09e3760, it's just nonsense changes to allow for testing and will be removed

**QA**
<img src="https://github.com/user-attachments/assets/74aa5c9d-4dc2-4f27-bea0-a5b512f22bd9" width="400" />

- As a learner, navigate to Learn > Home. You should see the banner with "You do not have enough storage for new resources. Ask your coach or administrator for help." message right away. Hear it being announced by the screen reader after all other initial announcements, such as Kolibri was loaded.
  - Note that the banner will be displayed under any circumstances, even wrong ones - just for the purpose of testing
  - Note that the screen reader will also announce the number of points if you have any in a way that is rather confusing, just saying 500 without any context (at least my Orca does this) - not in the scope of this PR.
- After 15 seconds, the banner message changes to "Updated storage notification banner message that should be announced by screen readers." This too should be announced.
- Also observe browser console, particularly see if you spot any `[useKLiveRegion] Could not send the message: ... ` error

Note that it's better to use screen reader to test this, rather than just observing

![Screenshot from 2024-07-23 22-17-19](https://github.com/user-attachments/assets/67c3e8b1-cedd-473d-a382-24f27ec14474)

because the content of region's elements gets cleared up really fast (deliberate and recommended behavior), making it nearly impossible to spot it.

- @radinamatic I also welcome your input on guidance on when to use live regions and polite/assertive messages https://deploy-preview-687--kolibri-design-system.netlify.app/usekliveregion/
----

## Testing checklist



## PR process

- [ ] ~IIf this is an important user-facing change, PR or related issue has a 'changelog' label~I
- [ ] ~IIf this includes an internal dependency change, a link to the diff is provided~I

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md
